EAD-Session-1-Assignment/functions.js

```diff
@@ -0,0 +1,373 @@
+		if(!(contact.length <= 10 && contact.length >= 8) ) {
+			this.errorDescription('required_e_contact',"Insert Number Between 8 to 10");
+			count++;
+		}else if(filterContact.test(contact)) {
+			this.errorDescription('required_e_contact',"Numeric values are only allowed");
+		} else {
+			this.removeDescription('required_e_contact',"");
+			count--;
+		}
+
+        //Validating user name
+		var filterName = new RegExp('\\d');
+		if(name.length <= 2) {
+			this.errorDescription('required_e_name',"Insert name greater then 2 word");
+			count++;
+		} else if(filterName.test(name)) {
+			this.errorDescription('required_e_name',"Numeric value not allowed");
+			count++;
+		} else{
+			this.removeDescription('required_e_name',"");
+			count--;
+		}
+
+        //Validating email
+		var filterEmail = new RegExp("^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$");
+  		if (!filterEmail.test(email)) {
+			this.errorDescription('required_e_email' ,"Insert a Valid Email");
+			count++;
+		} else {
+			this.removeDescription('required_e_email',"");
+			count--;
+		}
+
+        //Validating Password
+		var filterPassword = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})");
+		if(password.length <= 8){
+			this.errorDescription('required_e_pass' ,"Atleast 8 char required");
+			count++;
+		} else if (!filterPassword.test(password)) {
+			this.errorDescription('required_e_pass' ,"Insert Valid password format eg Abc@123");
+			count++;
+		} else if(password != confirmPassword) {
+			this.errorDescription('required_e_pass' ,"Both password should same");
+			count++;
+		} else {
+			this.removeDescription('required_e_pass',"");
+			count--;
+		}
+
+        // Validating Gender
+		if(gender === ""){
+			this.errorDescription('required_e_gender' ,"Select Any one Gender");
+			count++;
+		} else{
+			this.removeDescription('required_e_gender',"");
+			count--;
+		}
+
+		return count;
+    }
+
+    /**
+     * Method to validate the vehicle information
+     * @param {*} vehicleName is the vehicle name
+     * @param {*} type is the vehicle type
+     * @param {*} vehicleNumber is the vehicle number
+     * @param {*} identity is the vehicle identity
+     */
+    static validateVehicleInformation(vehicleName , type , vehicleNumber , identity) {
+        var count = 0;
+        //validating vehicle name
+        if(vehicleName.length < 3)
+        {
+			this.errorDescription('required_v_name',"Insert name greater then 2 word");
+			count++;
+        } else 
+        {
+			this.removeDescription('required_v_name',"");
+			count--;
+		}
+
+        //validating vehicle type
+		if(type == "none") {
+			this.errorDescription('required_v_type',"Select Type Of Your Vehicle");
+			count++;
+		} else {
+			this.removeDescription('required_v_type',"");
+			count--;
+		}
+
+        //validating vehicle number
+		if(vehicleNumber.length < 4) {
+			this.errorDescription('required_v_number',"Vehicle Number required");
```

Comment:
Change the error message like vehicle number length can not be less than 4 character. And also apply validation for vehicle number.

Reply:
Done